File: DSA/Python/src/dsa/lib/math/nums/tests/test_two_sum.py
from unittest import TestCase

from dsa.lib.math.nums.two_sum import TwoSum


class TestTwoSum(TestCase):
    def test_two_sum(self):
        self.assertEqual([0, 1], self.two_sum([2, 7, 11, 15], 9))
        self.assertEqual([1, 2], self.two_sum([3, 2, 4], 6))
        self.assertEqual([0, 1], self.two_sum([3, 3], 6))
        self.assertEqual([], self.two_sum([1, 2], 7))

    @staticmethod
    def two_sum(nums, target):
        return TwoSum.twoSum(nums, target)

File: DSA/Cpp/test/dsa/lib/math/nums/test_two_sum.cpp
#include <dsa/lib/math/nums/two_sum.hpp>
#include "nums_test.h"
namespace dsa::lib::math::nums {
namespace {

class TwoSumTest : public NumsTest {
 protected:
    static std::vector<int> twoSum(std::vector<int> &nums, int target) {
        return TwoSum::twoSum(nums, target);
    }
};
TEST_F(TwoSumTest, case1) {
    std::vector<int> nums = {2, 7, 11, 5};
    std::vector<int> actual = twoSum(nums, 9);
    std::vector<int> expected = {0, 1};
    EXPECT_EQ(expected, actual);
}
TEST_F(TwoSumTest, case2) {
    std::vector<int> nums = {3, 2, 4};
    std::vector<int> actual = twoSum(nums, 6);
    std::vector<int> expected = {1, 2};
    EXPECT_EQ(expected, actual);
}
TEST_F(TwoSumTest, case3) {
    std::vector<int> nums = {3, 3};
    std::vector<int> actual = twoSum(nums, 6);
    std::vector<int> expected = {0, 1};
    EXPECT_EQ(expected, actual);
}
TEST_F(TwoSumTest, case4) {
    std::vector<int> nums = {3, 3};
    std::vector<int> actual = twoSum(nums, 7);
    std::vector<int> expected = {};
    EXPECT_EQ(expected, actual);
}

}
}
